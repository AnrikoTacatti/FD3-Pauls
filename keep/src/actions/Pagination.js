import { PAGE } from '../stores/const.js';

export function actionPageNumber(dispatch, currentPage) {
    dispatch({ type: PAGE, data: { currentPage: currentPage } });
}






