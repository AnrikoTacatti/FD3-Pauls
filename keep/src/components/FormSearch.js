import React from 'react';
import { connect } from 'react-redux';
import { SEARH_TEXT } from '../stores/const.js';
import { withRouter } from "react-router";
class FormSearch extends React.PureComponent {
    constructor(props) {
        super(props);
    }
    Sarch = () => {

        let data = { text: this.inputSarch.value };
        this.props.dispatch({ type: SEARH_TEXT, data });
        this.props.history.push("/serch");

    }
    render() {
        return (
            <React.Fragment>
                <div className="form-search">
                    <div className="form-inner">
                        <input type="text" className="task-list-name" name="phone" placeholder="Search..." required="" ref={c => this.inputSarch = c} />
                        <button type="button" className="form-submit btn" onClick={this.Sarch} >Sarch</button>
                    </div>
                </div>

            </React.Fragment>
        );

    }

}



export default connect()(withRouter(FormSearch));






