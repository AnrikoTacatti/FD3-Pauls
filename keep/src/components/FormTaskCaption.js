import React from 'react';
import { connect } from 'react-redux';
import {
    actioncloseFormNewTaskCaption,
    actioncloseFormEditTaskCaption,
    actionsetNewTaskCaption,
    actioneditTaskCaption
} from '../actions/FormTaskCaption.js';



export class FormTaskCaption extends React.PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            /* TaskLists: this.props.stateTaskLists.TaskLists*/
            openFormNewTaskCaption: false,
            openFormEditTaskCaption: false
        }
    }

    UNSAFE_componentWillReceiveProps(newProps) {
        this.setState({ openFormNewTaskCaption: newProps.openFormNewTaskCaption });
        this.setState({ openFormEditTaskCaption: newProps.openFormEditTaskCaption });
    }

    closeFormNewTaskCaption = () => {
        let data = false;
        actioncloseFormNewTaskCaption(data, this.props.dispatch)
    }

    closeFormEditTaskCaption = () => {
        let data = { active: false };
        actioncloseFormEditTaskCaption(data, this.props.dispatch);
    }

    addNewTaskCaption = () => {
        let data = { title: this.inputTaskListName.value };
        actionsetNewTaskCaption(data, this.props.dispatch);
    }

    editTaskCaption = () => {
        let data = { title: this.inputTaskListName.value, keychapter: this.state.openFormEditTaskCaption.keychapter };
        actioneditTaskCaption(data, this.props.dispatch);
    }


    render() {
        // console.log("FormTaskCaption", this.props);

        return (
            <React.Fragment>
                {this.state.openFormEditTaskCaption.active === true && <React.Fragment>
                    <div className="form-add-task-caption form">
                        <div className="form-inner">
                            <p className="form-title">Edit task list</p>
                            <input type="text" className="task-list-name" name="title" defaultValue={this.state.openFormEditTaskCaption.name} placeholder="Enter task list name" required="" ref={c => this.inputTaskListName = c} />
                        </div>
                        <div className="form-add-task-caption__footer">
                            <button type="button" title="Cancel" className="form-cancel btn" onClick={this.closeFormEditTaskCaption}>Cancel</button>
                            <button type="button" className="form-submit btn" onClick={this.editTaskCaption} >Submit</button>
                        </div>
                    </div>
                    <div className="form-bg"></div>
                </React.Fragment>
                }
                {this.state.openFormNewTaskCaption === true && <React.Fragment>
                    <div className="form-add-task-caption form">
                        <div className="form-inner">
                            <p className="form-title">Add task list</p>
                            <input type="text" className="task-list-name" name="phone" placeholder="Enter task list name" required="" ref={c => this.inputTaskListName = c} />
                        </div>
                        <div className="form-add-task-caption__footer">
                            <button type="button" title="Cancel" className="form-cancel btn" onClick={this.closeFormNewTaskCaption}>Cancel</button>
                            <button type="button" className="form-submit btn" onClick={this.addNewTaskCaption} >Submit</button>
                        </div>
                    </div>
                    <div className="form-bg"></div>
                </React.Fragment>
                }
            </React.Fragment>
        );

    }

}

const mapStateToProps = function (state) {
    return {
        // весь раздел Redux state под именем counters будет доступен
        // данному компоненту как this.props.counters
        openFormNewTaskCaption: state.stateTaskLists.openFormNewTaskCaption,
        openFormEditTaskCaption: state.stateTaskLists.openFormEditTaskCaption,
    };
};

export default connect(mapStateToProps)(FormTaskCaption);






