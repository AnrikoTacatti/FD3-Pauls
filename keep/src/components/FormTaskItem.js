import React from 'react';
import { connect } from 'react-redux';
import {
    actioncloseFormNew,
    actioncloseFormEdit,
    actionsetNewTaskItem,
    setTaskItem
} from '../actions/FormTaskItem.js';

export class FormTaskItem extends React.PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            openFormTaskItemEdit: false,
            openFormTaskItemNew: false,
            inputTaskItemNameRequired: true,
            inputTaskItemTextRequired: true,
        };
        this.inputTaskItemName = React.createRef();
        this.inputTaskItemText = React.createRef();

    }

    UNSAFE_componentWillReceiveProps(newProps) {
        this.setState({ openFormTaskItemEdit: newProps.openFormTaskItemEdit });
        this.setState({ openFormTaskItemNew: newProps.openFormTaskItemNew });

    }

    closeFormNew = () => {
        let data = { active: false };
        actioncloseFormNew(data, this.props.dispatch);
    }


    closeFormEdit = () => {
        let data = { active: false };
        actioncloseFormEdit(data, this.props.dispatch);
    }

    addNewTaskItem = () => {
        if (this.inputTaskItemName.current.value == "") {
            this.setState({ inputTaskItemNameRequired: false });
        }
        else {
            this.setState({ inputTaskItemNameRequired: true });
        }
        if (this.inputTaskItemText.current.value == "") {
            this.setState({ inputTaskItemTextRequired: false });
        }
        else {
            this.setState({ inputTaskItemTextRequired: true });
        }
        if (this.inputTaskItemName.current.value != "" && this.inputTaskItemText.current.value != "") {
            let data = { keychapter: this.state.openFormTaskItemNew.keychapter, title: this.inputTaskItemName.current.value, text: this.inputTaskItemText.current.value };
            actionsetNewTaskItem(data, this.props.dispatch);
        }
    }

    editTaskItem = () => {
        let data = { title: this.inputTaskItemName.value, text: this.inputTaskItemText.value, keychapter: this.state.openFormTaskItemEdit.keychapter, keyitem: this.state.openFormTaskItemEdit.keyitem };
        setTaskItem(data, this.props.dispatch);
    }

    render() {
        //  console.log("FormTaskItem", this.props);
        // console.log("FormTaskItem", this.state);
        return (
            <React.Fragment >
                {
                    this.state.openFormTaskItemNew.activ === true && <React.Fragment>
                        <div className="form-add-task-caption form">
                            <div className="form-inner">
                                <p className="form-title">Add task item</p>
                                <input type="text" className={`task-item-name ${this.state.inputTaskItemNameRequired == false ? "error" : ""}`} name="title" placeholder="Enter task item name" required="" ref={this.inputTaskItemName} />
                                <textarea className={`task-item-text ${this.state.inputTaskItemTextRequired == false ? "error" : ""}`} name="text" cols="60" rows="5" placeholder="Enter task item text" required="" ref={this.inputTaskItemText} />
                            </div>
                            <div className="form-add-task-caption__footer">
                                <button type="button" title="Cancel" className="form-cancel btn" onClick={this.closeFormNew}>Cancel</button>
                                <button type="button" className="form-submit btn" onClick={this.addNewTaskItem} >Submit</button>
                            </div>
                        </div>
                        <div className="form-bg"></div>
                    </React.Fragment>
                }
                {
                    this.state.openFormTaskItemEdit.activ === true && <React.Fragment>
                        <div className="form-add-task-caption form">
                            <div className="form-inner">
                                <p className="form-title">Edit task item</p>
                                <input type="text" className="task-item-name" defaultValue={this.state.openFormTaskItemEdit.name} name="name" placeholder="Enter task item name" required="" ref={c => this.inputTaskItemName = c} />
                                <textarea className="task-item-text" defaultValue={this.state.openFormTaskItemEdit.text} name="text" cols="60" rows="5" placeholder="Enter task item text" required="" ref={c => this.inputTaskItemText = c} />
                            </div>
                            <div className="form-add-task-caption__footer">
                                <button type="button" title="Cancel" className="form-cancel btn" onClick={this.closeFormEdit}>Cancel</button>
                                <button type="button" className="form-submit btn" onClick={this.editTaskItem} >Submit</button>
                            </div>
                        </div>
                        <div className="form-bg"></div>
                    </React.Fragment>
                }
            </React.Fragment >
        );

    }

}

const mapStateToProps = function (state) {
    return {
        // весь раздел Redux state под именем counters будет доступен
        // данному компоненту как this.props.counters
        openFormTaskItemEdit: state.stateTaskLists.openFormTaskItemEdit,
        openFormTaskItemNew: state.stateTaskLists.openFormTaskItemNew
    };
};

export default connect(mapStateToProps)(FormTaskItem);






