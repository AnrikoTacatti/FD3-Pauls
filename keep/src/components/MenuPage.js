import React from 'react';
import { NavLink } from 'react-router-dom';
import { icoHome, icoAbout } from '../ico/ico.js';

class MenuPage extends React.Component {
    render() {
        return (
            <React.Fragment>
                <li><NavLink to="/" exact className="PageLink" activeClassName="ActivePageLink"> {icoHome()} Главная </NavLink></li>
                <li> <NavLink to="/about" className="PageLink" activeClassName="ActivePageLink"> {icoAbout()} О компании</NavLink></li>
            </React.Fragment>
        );
    }
}

export default MenuPage;




