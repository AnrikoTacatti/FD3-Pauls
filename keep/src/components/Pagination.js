"use strict";
import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { withRouter } from "react-router";
import { actionPageNumber } from '../actions/Pagination.js';

export class Pagination extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            currentPage: this.props.location.search != "" ? Number(this.props.location.search.match(/\d+/g)) : "all",
        }
    }

    UNSAFE_componentWillReceiveProps = (newProps) => {
        this.setState((prevState) => { return { currentPage: newProps.currentPage === null ? prevState.currentPage : newProps.currentPage } });
    }

    render() {
        //  console.log("Pagination", this.props);
        const pageNumber = []
        for (let i = 1; i <= Math.ceil(this.props.total / this.props.perpage); i++) {
            pageNumber.push(i);
        }
        return (
            <ul className="pagination">
                {pageNumber.map((number) => <li key={number} className={number == this.state.currentPage ? "active" : ""} >
                    <NavLink to={this.props.location.pathname + "?page_" + number} onClick={actionPageNumber.bind(null, this.props.dispatch, number)} >{number}</NavLink></li>)}
                <li className={"all" === this.state.currentPage ? "active" : ""} ><NavLink to={this.props.location.pathname} onClick={actionPageNumber.bind(null, this.props.dispatch, "all")}>Все</NavLink></li>
            </ul>
        );

    }
}

const mapStateToProps = function (state) {
    return {
        // весь раздел Redux state под именем counters будет доступен
        // данному компоненту как this.props.counters
        currentPage: state.stateTaskLists.currentPage,
    };
};

export default connect(mapStateToProps)(withRouter(Pagination));






