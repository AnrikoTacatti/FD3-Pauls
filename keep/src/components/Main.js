"use strict";
import React from 'react';
import FormTaskItem from './FormTaskItem.js';
import TaskItemContainer from './TaskItemContainer.js';
import { icoSave } from '../ico/ico.js';
import { actionsaveBook } from '../actions/Main.js';
class Main extends React.PureComponent {
    constructor(props) {
        super(props);
    }

    forTaskListschild = () => {
        let tasklistitem = [];
        let listtask = this.props.TaskListsItemsSort;
        let length = 0;

        switch (true) {
            case listtask.length >= 4: length = 4; break;
            case listtask.length >= 3: length = 3; break;
            case listtask.length >= 2: length = 2; break;
            case listtask.length >= 1: length = 1; break;
        }

        for (let i = 0; i < length; i++) {
            let keyitem = listtask[i].key;
            let keychapter = listtask[i].keychapter;
            tasklistitem.push(
                <TaskItemContainer data={listtask[i]} keychapter={keychapter} keyitem={keyitem} key={keyitem} index={i + 1}
                    attrdata={this.props.locationPathname !== undefined ? this.props.locationPathname : "all"}
                />
            )
        }

        return tasklistitem;
    }

    forPinTaskListschild = () => {
        let tasklistitem = [];
        let listtask = [];
        if (this.props.TaskListsItemsSort.length) this.props.TaskListsItemsSort.forEach(function (el, index, array) { el.pin === true && listtask.push(el) });
        let length = 0;

        switch (true) {
            case listtask.length >= 4: length = 4; break;
            case listtask.length >= 3: length = 3; break;
            case listtask.length >= 2: length = 2; break;
            case listtask.length >= 1: length = 1; break;
        }

        for (let i = 0; i < length; i++) {
            let keyitem = listtask[i].key;
            let keychapter = listtask[i].keychapter;
            tasklistitem.push(
                <TaskItemContainer data={listtask[i]} keychapter={keychapter} keyitem={keyitem} key={keyitem} index={i + 1}
                    attrdata={this.props.locationPathname !== undefined ? this.props.locationPathname : "all"}
                />
            )
        }
        return tasklistitem;
    }


    render() {
        console.log("render Main");
        return (
            <React.Fragment>
                <div className="task-chapter task-chapter--main" >
                    <style>
                        {` .task-item{ -webkit-animation-name: none;
                                animation-name: none;} 
                            .${this.props.locationPathname !== undefined ? this.props.locationPathname : "all"} { 
                                -webkit-animation-name: slideInUp;
                                animation-name: slideInUp;
                                opacity: 0; }
                            `
                        }
                    </style>
                    <div className="task-chapter__header">
                        <div className="task-chapter__title">
                            Последние задачи
                        </div>
                    </div>
                    <div className="task-item-list">
                        {this.forTaskListschild()}
                    </div>
                    <div className="task-chapter__header">
                        <div className="task-chapter__title">
                            Закрепленные задачи
                        </div>
                    </div>
                    <div className="task-item-list">
                        {this.forPinTaskListschild()}
                    </div>
                    <div className="task-chapter__header">
                        <div className="task-chapter__title">
                            Записная книжка
                        </div>
                    </div>
                    <div className="book" style={{ "animationDelay": `1s` }}>
                        <div className="book__title">Записная книжка <span className="book__save" onClick={this.props.cbsaveBook}> {icoSave()} </span></div>
                        {console.log("Записная книжка")}
                        {this.props.Book !== undefined && <textarea className="book__text" ref={this.props.cbsetNewTextRef} defaultValue={this.props.Book !== undefined && this.props.Book.text} placeholder="Запишите что-нибудь"></textarea>}
                    </div>

                </div>
                <FormTaskItem />
            </React.Fragment>

        );
    }


}

export default Main;
