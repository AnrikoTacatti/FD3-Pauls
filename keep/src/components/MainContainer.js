"use strict";
import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from "react-router";
import FormTaskItem from './FormTaskItem.js';
import Main from './Main.js';
import api from '../api/api.js';

class MainContainer extends React.PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            TaskListsItemsSort: this.props.stateTaskListsItemsSort,
            locationPathname: this.props.match.params.chapter,
            Book: this.props.Book
        }

    }

    UNSAFE_componentWillReceiveProps = (newProps) => {
        this.setState({ TaskListsItemsSort: newProps.stateTaskListsItemsSort });
        this.setState({ locationPathname: newProps.match.params.chapter });
        this.setState({ Book: newProps.Book });
    }

    saveBook = () => {
        let data = { text: this.bookText.value };
        api.saveBook(data, this.props.dispatch);

    }
    setNewTextRef = (ref) => {
        this.bookText = ref;
    };

    render() {
        console.log("render MainContainer");
        return (
            <React.Fragment>
                <Main {...this.state} cbsetNewTextRef={this.setNewTextRef} cbsaveBook={this.saveBook} />
                <FormTaskItem />
            </React.Fragment>
        );
    }


}

const mapStateToProps = function (state) {
    return {
        // весь раздел Redux state под именем counters будет доступен
        // данному компоненту как this.props.counters
        stateTaskListsItemsSort: state.stateTaskLists.TaskListsItemsSort,
        Book: state.stateTaskLists.Book
    };
};

export default connect(mapStateToProps)(withRouter(MainContainer));
