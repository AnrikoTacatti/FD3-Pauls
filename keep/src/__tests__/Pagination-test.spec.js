"use strict";
import React from 'react';
import { Pagination } from '../components/Pagination';
import { NavLink } from 'react-router-dom';
jest.mock('../actions/Pagination.js');
import { actionPageNumber } from '../actions/Pagination.js';



describe('Search-test', () => {
    let state = {
        location: {
            "search": { "math": "" },
            "pathname": "/chapter"
        }
    }

    let dispatch = jest.fn();
    actionPageNumber.mockImplementation((dispatch, data) => { return data; });


    const component = shallow(
        <Pagination perpage="10" total={35} location={location} dispatch={dispatch} />
    );

    test('Search Snapshot', () => {
        expect(component).toMatchSnapshot();
    });
    test('Pagination length', () => {
        const link = component.find(NavLink);
        expect(link).toHaveLength(5);
    });
    test('Pagination action', () => {
        const link = component.find(NavLink).at(0);
        link.simulate('click');
        expect(actionPageNumber).toHaveBeenCalledWith(dispatch, 1);

    });

});



