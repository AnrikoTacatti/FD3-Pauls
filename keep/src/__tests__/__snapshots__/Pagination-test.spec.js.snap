// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search-test Search Snapshot 1`] = `
<ul
  className="pagination"
>
  <li
    className=""
    key="1"
  >
    <NavLink
      onClick={[Function]}
      to="/?page_1"
    >
      1
    </NavLink>
  </li>
  <li
    className=""
    key="2"
  >
    <NavLink
      onClick={[Function]}
      to="/?page_2"
    >
      2
    </NavLink>
  </li>
  <li
    className=""
    key="3"
  >
    <NavLink
      onClick={[Function]}
      to="/?page_3"
    >
      3
    </NavLink>
  </li>
  <li
    className=""
    key="4"
  >
    <NavLink
      onClick={[Function]}
      to="/?page_4"
    >
      4
    </NavLink>
  </li>
  <li
    className="active"
  >
    <NavLink
      onClick={[Function]}
      to="/"
    >
      Все
    </NavLink>
  </li>
</ul>
`;
